Sources/LeafKit/LeafLexer/LeafParameterTypes.swift

```diff
@@ -50,30 +50,111 @@ public enum Parameter: Equatable, CustomStringConvertible {
+    case not = "!"              //    X                       X
+    case equal = "=="           //    X                 X
+    case unequal = "!="         //    X                 X
+    case greater = ">"          //    X                 X
+    case greaterOrEqual = ">="  //    X                 X
+    case lesser = "<"           //    X                 X
+    case lesserOrEqual = "<="   //    X                 X
+    case and = "&&"             //    X                 X
+    case or = "||"              //    X                 X
+    //   Mathematical Calcs     // -----------------------------------------
+    case plus = "+"             //          X           X
+    case minus = "-"            //          X     X     X     X
+    case divide = "/"           //          X           X
+    case multiply = "*"         //          X           X
+    case modulo = "%"           //          X           X
+    //   Assignment/Existential //
+    case assignment = "="       //                X     X
+    case nilCoalesce = "??"     //                X     X
+    case evaluate = "`"         //                X           X
+    //   Scoping
+    case scopeRoot = "$"        //                            X           X
+    case scopeMember = "."      //                      X                 X
+    case subOpen = "["          //                      X                 X
+    case subClose = "]"         //                                  X     X
     
     /// Raw string value of the operator - eg `!=`
     public var description: String { return rawValue }
+    
+    // MARK: Internal Only
+    
+    // State booleans
+    internal var logical: Bool { Self.states["logical"]!.contains(self) }
+    internal var mathematical: Bool { Self.states["mathematical"]!.contains(self) }
+    internal var existential: Bool { Self.states["existential"]!.contains(self) }
+    internal var scoping: Bool { Self.states["scoping"]!.contains(self) }
+    
+    internal var unaryPrefix: Bool { Self.states["unaryPrefix"]!.contains(self) }
+    internal var unaryPostfix: Bool { Self.states["unaryPostfix"]!.contains(self) }
+    internal var infix: Bool { Self.states["unaryPostfix"]!.contains(self) }
+    
+    internal var available: Bool { !Self.states["unavailable"]!.contains(self) }
+    
+    internal static let precedenceMap: [(check: ((LeafOperator) -> Bool), infixed: Bool)] = [
+        (check: { $0 == .not }, infixed: false), // unaryNot
+        (check: { $0 == .multiply || $0 == .divide }, infixed: true), // Mult/Div
+        (check: { $0 == .plus || $0 == .minus }, infixed: true), // Plus/Minus
+        (check: { $0 == .greater || $0 == .greaterOrEqual }, infixed: true), // >, >=
+        (check: { $0 == .lesser || $0 == .lesserOrEqual }, infixed: true), // <, <=
+        (check: { $0 == .equal || $0 == .unequal }, infixed: true), // !, !=
+        (check: { $0 == .and || $0 == .or }, infixed: true), // &&, ||
+    ]
```

Comment:
Needs a little work but the same precedence map introduced a couple versions ago, in descending order of precedence for assisting in resolving ambiguous expressions that aren't `(lhs infix rhs), (lhs postfix) (prefix rhs)` and assisting in wrapping them into the correct order, if possible.

moved over from LeafLexer